doc/wiki/dashboard-audit.md

```diff
@@ -0,0 +1,672 @@
+AND "medical_histories"."patient_id" = "patients"."id"
+WHERE "patients"."deleted_at" IS NULL
+  AND "patients"."assigned_facility_id" = $1
+  AND "medical_histories"."deleted_at" IS NULL
+  AND "medical_histories"."hypertension" = $2
+  AND "patients"."status" != $3
+  AND ("patients"."recorded_at" IS NOT NULL)
+GROUP BY DATE_TRUNC('month', "patients"."recorded_at"::timestamptz AT TIME ZONE 'ASIA/KOLKATA') AT TIME ZONE 'ASIA/KOLKATA' [["assigned_facility_id", "fdbeb339-770f-4021-a81a-44f9bc9a5e79"],["hypertension", "yes"],["status", "dead"]]
+```
+
+**Where it's shown in the Dashboard**
++ **Reports:** Overview, Details, Cohort reports
++ **Home:** BP controlled, BP not controlled, Missed visits
++ **Simple App:** Progress tab
+
+**Things we should fix**
++ [ ] Make copy definition more specific and more closely match how the query is calculated
++ [ ] In `reports.js` rename `data-registrations` -> `assigned-patients`
++ [ ] Anything that grabs data from `dashboard_analytics` could display different numbers from the main Reports pages because of differences in when data is cached
++ [ ] Review `UserAnalytics` code paths
+
+---
+
+## BP controlled
+**Copy used in the Dashboard**
++ Numerator: Patients with BP <140/90 at their last visit in the last 3 months
++ Denominator: Hypertension patients assigned to `region_name`, registered before the last 3 months. Dead and lost to follow-up patients are excluded.
+
+**How it's calculated**
+The number of patients assigned to a facility registered before the last 3 months where the patient:
++ Is hypertensive
++ Is not dead
++ Has a BP measure taken within the last 3 months
++ Last BP measure taken is <140/90
+
+*Sample SQL query*
+```sql
+SELECT COUNT(*)
+FROM
+  (SELECT DISTINCT ON (latest_blood_pressures_per_patient_per_months.patient_id) *
+   FROM "latest_blood_pressures_per_patient_per_months"
+   WHERE (medical_history_hypertension = 'yes')
+     AND "latest_blood_pressures_per_patient_per_months"."patient_status" != $1
+     AND "latest_blood_pressures_per_patient_per_months"."assigned_facility_id" = $2
+     AND (patient_recorded_at <= '2019-07-31 18:29:59.999999')
+     AND (bp_recorded_at >= '2019-07-31 18:30:00'
+          AND bp_recorded_at <= '2019-10-31 18:29:59.999999')
+   ORDER BY latest_blood_pressures_per_patient_per_months.patient_id,
+            bp_recorded_at DESC, bp_id) latest_blood_pressures_per_patient_per_months
+WHERE (systolic < 140
+       AND diastolic < 90) [["patient_status", "dead"],["assigned_facility_id", "399c52a8-4b49-41d4-8704-cc3985ac26e6"]]
+```
+
+**Where it's shown in the Dashboard**
++ **Reports:** Overview
++ **Home:** BP controlled
++ **Simple App:** Progress tab
+
+**Things we should fix**
++ [ ] The "Home" and "Reports" pages in the Dashboard use `ControlRateQuery` and the "Progress tab" in the Simple App uses `ControlRateService`. We need to look into these queries and consolidate differences.
```

Comment:
These are not different queries - the Service uses the underlying Query, so things are all good here.